File: python_backend/id_assign_/rent_ids.py
import copy
import random
from id_assign_.ids_for_rent import _ids_for_rent
from pprint import pprint


def rent_ids(id_dict, custom_names, ignore=[]):
    """extend id_dict by rentring available assign_ids for custom_names"""
    ids = copy.deepcopy(_ids_for_rent)
    random.shuffle(ids)
    for name in custom_names:
        if name in ignore:
            continue
        id_ = ids.pop(0)
        id_dict[id_] = name
    return id_dict


if __name__ == "__main__":
    pprint(rent_ids({'b': 'Duke', 'q': 'Jester'}, ['Joker', 'Morty']))

File: app (1)/src/apiHelpers/idAssign/rentIds.js
import {idsForRent} from "./idsForRent";
import {shuffle} from "../../Components/helpers/shuffleArray";


export function rentIds(idDict, customNames, ignore) {
    /**for custom pieces not for.includes(subbing) a standard (these are pawn promotion options), pick id from the 20 alphabet
       characters not used by a standard piece.
    */
    var id;
    var ids = JSON.parse(JSON.stringify(idsForRent))
    ids = shuffle(ids)
    for (var name of customNames) {
        if (ignore && ignore.includes(name)) {
            continue
        }
        id = ids.shift();
        idDict[id] = name;
    }
    return idDict;
}

// module.exports = rentIds;

/** for node.js
if (require.main === module) {
    console.log(rentIds({'b': 'Duke', 'q': 'Jester'}, ['Joker', 'Morty']))

}
*/